conf/find_invalid_waivers.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python3
+"""
+This is a standalone script which processes provided results.txt.gz files
+and identifies invalid waivers. The waiver is invalid if it:
+ - did not match any test results,
+ - or only matched the 'pass' test results.
+
+The identified invalid waivers are printed to the standard output.
+"""
+
+import re
+import sys
+import gzip
+import pathlib
+import argparse
+import textwrap
+
+# add the parent directory to the sys.path so we can import from the lib directory
+sys.path.append(str(pathlib.Path(__file__).resolve().parent.parent))
+from lib import waive, versions
+
+_sections_cache = None
```

Comment:
I don't think this global variable is needed.

It exists in `lib.waive` because we don't want passing it around in test code, we want the test to be able to just call
```python
results.report(...)
```
without passing some parsed list of waivers around its logic.

But that problem doesn't exist here in your code, so you can IMHO leave it as a normal variable inside `main()`.